packages/webapp/src/components/Task/TaskLocations/index.jsx

```diff
@@ -55,10 +55,13 @@ export default function PureTaskLocations({
   );
 
   const onSelectLocation = (location_id) => {
-    setValue(
-      LOCATIONS,
-      isMulti ? getSelectedLocations(location_id, selectedLocations) : [{ location_id }],
-    );
+    if(!isMulti && getValues('show_wild_crop')){
```

Comment:
use early return 
```
const onSelectLocation = (location_id) => {
  if(!isMulti && getValues('show_wild_crop')) return;
  setValue(
    LOCATIONS,
    isMulti ? getSelectedLocations(location_id, selectedLocations) : [{ location_id }],
  );
};
```

Reply:
@anant00729 Good catch! Can you check this again. 

I instead handled the !multi condition for the wild crop here. Now when a location is selected and the checkbox is also selected it will now deselect the checkbox and select the area to fit in with how the map multiselect works.